app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -694,6 +694,18 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                             currentInfo.getOriginalUrl());
                 }
                 return true;
+            case R.id.menu_item_share_stream:
+                if (currentInfo != null) {
+                    Stream stream;
```

Comment:
I think this could be `final`

Reply:
Nope, its reference is changed below. Here the reference is null, and after it will be either poiting on an object return by `getDefaultAudioStream()` or `getSelectedVideoStream()`

Reply:
@B0pol: No, it isn't assigned here yet, and it's afterwards only assigned once: in `if` or in `else`. (With e.g. `try`-`catch` it would've been different.)

Reply:
Yeah it is never assigned to null, it's only assigned in two different branches.

I also removed the `@Nullable` annotation on `getDefaultAudioStream()` because it cannot be null. If there's no audio stream or no appropriate format, it would fail by calling `get(-1)` eventually and throwing an index out of bounds exception.

And while theoretically `getSelectedVideoStream()` could return null, given that we only call it inside this conditional, it would never happen. But the IDE still complains 🤷 . An assertion would silence it... I don't know if it's worth it.